src/librustc_trans/mir/constant.rs

```diff
@@ -706,7 +704,11 @@ impl<'a, 'tcx> MirConstContext<'a, 'tcx> {
 
                 let base = match tr_lvalue.base {
                     Base::Value(llval) => {
-                        let align = type_of::align_of(self.ccx, ty);
+                        let align = if type_is_sized(self.ccx.tcx(), ty) {
```

Comment:
I don't understand this code.  I think this was proposed by @eddyb.  Can this just be fixed while we're looking at it and, if so, how?